common/src/main/java/org/astraea/common/admin/Admin.java

```diff
@@ -282,6 +282,19 @@ default CompletionStage<Set<String>> idleTopic(List<TopicChecker> checkers) {
 
   CompletionStage<Void> moveToFolders(Map<TopicPartitionReplica, String> assignments);
 
+  /**
+   * Declare the preferred data folder for the designated partition at a specific broker.
+   *
+   * <p>Preferred data folder is the data folder a partition will use when it is being placed on the
+   * specific broker({@link Admin#moveToBrokers(Map)}). This API won't trigger a folder-to-folder
+   * replica movement. To Perform folder-to-folder movement, consider use {@link
+   * Admin#moveToFolders(Map)}.
+   *
+   * <p>This API is not transactional. It won't work properly when there are concurrent changes to
```

Comment:
`It may alter folders if the target broker has out-of-date metadata or running reassignments`